frontend/fingertips-frontend/lib/escapeString.test.ts

```diff
@@ -0,0 +1,50 @@
+import { escapeString } from './escapeString';
+
+describe('escapeString', () => {
+  it('should escape special characters correctly', () => {
+    const input = '+-&|!(){}[]^"~*?:\\/';
```

Comment:
Why two backslashes?
```suggestion
    const input = '+-&|!(){}[]^"~*?:\/';
```

Reply:
The first backslash is used to escape the second backslash. In JavaScript strings, the backslash has a special meaning, I believe.

You can test it here: https://regex101.com/.
\\ means "I want to match a single backslash (\)".
However, if I write it as \/, it match anything  /. because the first \ is use to escape the \ to appear.  and that is what we are looking for. 

Normally, we don't need to escape /, but because we're using replace instead of directly using a regular expression which is a string type we need to escape with it  / 
 So to be able to match \/ we have to do this \\\/ .

Reply:
Yeah, makes sense now I look at it again.